packages/core/helper-plugin/src/utils/auth.ts

```diff
@@ -0,0 +1,196 @@
+// TODO @soupette we need to refactor this file
+
+import isEmpty from 'lodash/isEmpty';
+import isNil from 'lodash/isNil';
+
+const TOKEN_KEY = 'jwtToken';
+const USER_INFO = 'userInfo';
+const CURRENT_STEP = 'GUIDED_TOUR_CURRENT_STEP';
+const COMPLETED_STEPS = 'GUIDED_TOUR_COMPLETED_STEPS';
+const SKIPPED = 'GUIDED_TOUR_SKIPPED';
+const THEME_KEY = 'STRAPI_THEME'; // Also used in packages/core/admin/admin/src/components/ThemeToggleProvider/index.js
+const UPLOAD_MODAL_VIEW = 'STRAPI_UPLOAD_MODAL_VIEW';
+const UPLOAD_VIEW = 'STRAPI_UPLOAD_LIBRARY_VIEW';
+
+interface UserInfo {
+  email: string;
+  firstname: string;
+  lastname?: string;
+  username?: string;
+  preferedLanguage?: string;
+  id: number;
+  isActive?: boolean;
+  blocked: boolean;
+  createdAt: Date;
+  updatedAt: Date;
+}
+
+interface StorageItems {
+  userInfo: UserInfo;
+  jwtToken: string;
+  STRAPI_THEME: 'light' | 'dark';
+  GUIDED_TOUR_CURRENT_STEP: string | null;
+  GUIDED_TOUR_COMPLETED_STEPS: string[] | null;
+  GUIDED_TOUR_SKIPPED: boolean | null;
+  STRAPI_UPDATE_NOTIF: boolean | null;
+  STRAPI_UPLOAD_MODAL_VIEW: 0 | 1 | null; // grid or list view
+  STRAPI_UPLOAD_LIBRARY_VIEW: 0 | 1 | null; // grid or list view
+  // FIXME: No idea what this is or what its value is
+  videos: string | null;
+  // FIXME: No idea what this is or what its value is
+  onboarding: string | null;
+}
+
+type StorageItemValues = StorageItems[keyof StorageItems];
+
+const parse = JSON.parse;
+const stringify = JSON.stringify;
+
+const auth = {
+  clear(key: keyof StorageItems) {
+    if (localStorage && localStorage.getItem(key)) {
```

Comment:
I do not, we do the check in a lot of places and I agree it doesn't seem necessary. I removed some and tested and it seems fine. I'm a little hesitant to remove all of them though, what do you think?

Reply:
I would assume this was for server environments, is the code used on the server at all? 🤔 If we're not, then i would remove it.

Reply:
At least on safari it looks like the app will crash if a user selects the "Block all cookies" option. But even the check for localStorage will cause a crash. I don't know if we need to support this or not. If we do I think it's another ticket/pr, but for this one I'll just remove the checks since it doesn't make a difference.